### PR TITLE
feat: transform code to array payload

### PR DESCRIPTION
**Pull request by damasch**
remove parameter
convert to array
rewrite the __get() method and access the payload array add unit tests

### CODE REVIEW

File: src/Transaction.php [hunk 1]
```diff
@@ -5,69 +5,54 @@
...
+        if (count($this->requiredKeys) < count($payload)) {
+            throw new \InvalidArgumentException("Too many keys in payload");
+        }
 
-    /**
-     * @var string The checkout token.
-     */
-    private $checkout_token;
+        foreach ($this->requiredKeys as $key) {
+            if (!array_key_exists($key, $payload)) {
```
Comment (pascaliske) [inline src/Transaction.php:39]: What if the key is available but empty? We should catch that as well, imo...

File: src/Transaction.php [hunk 2]
```diff
@@ -78,19 +63,10 @@ public function __get($property) {
...
-            'url' => $this->url,
-            'name' => $this->name,
-            'articleEID' => $this->external_article_id,
-            'magazineKey' => $this->magazine_key,
-            'checkoutToken' => $this->checkout_token,
-        ];
         $config = new Config();
         $url = sprintf("%s/transaction/verify", $config->get("origins", "api_url"));
-        $response = new NewsLadderRequest($url, $payload);
+        $response = new NewsLadderRequest($url, $this->payload);
```
Comment (pascaliske) [inline src/Transaction.php:68]: We renamed the Request class earlier. We need toadjust it here. 😊

File: src/Transaction.php [hunk 3]
```diff
@@ -5,69 +5,54 @@
...
+    public function __construct(array $payload) {
+        if (empty($payload)) {
+            throw new \InvalidArgumentException("Payload is empty");
+        }
 
-    /**
-     * @var string The magazine key.
-     */
-    private $magazine_key;
+        if (count($this->requiredKeys) < count($payload)) {
```
Comment (pascaliske) [inline src/Transaction.php:34]: Is this necessary? We could just ignore additional keys as those aren't used anyways...

**Review by pascaliske — CHANGES_REQUESTED** [review]
Hi @damasch,

I left you some minor questions about the payload checks. 😊

File: src/Transaction.php (hunk 3 above)
Comment (damasch) [reply src/Transaction.php:34]: But it is in the payload to the Service.

File: src/Transaction.php (hunk 1 above)
Comment (damasch) [reply src/Transaction.php:39]: We could Check against a regex pattern.

Comment (pascaliske) [reply src/Transaction.php:39]: I think a simple `isset($this->requiredKeys[$key] && !empty($this->requiredKeys[$key])` is sufficient for the time being...

File: src/Transaction.php (hunk 3 above)
Comment (pascaliske) [reply src/Transaction.php:34]: Yeah, then we should filter the payload by the required keys:

```php
$this->payload = array_intersect_key($payload, array_flip($this->requiredKeys));
```

**Review by pascaliske — APPROVED** [review]
💪🏻

**Pull request merged**